### DISCUSSION TITLE
Provide support for metrics from Azure Logs

### DISCUSSION THREAD

**Discussion by locmai** [tomkerkhove/promitor#1975] [Ideas]
## Context

We are using the [Container insights](https://docs.microsoft.com/en-us/azure/azure-monitor/containers/container-insights-overview) from Azure to collect and store the performance metrics from the AKS Windows containers. Some of those went into the `Insights > Logs` instead.

The Microsoft folks recommended us to get the metrics from there. As an example, the query for the average of all containers will look like this:

```
Perf
| where ObjectName == "K8SContainer" and CounterName == "cpuUsageNanoCores"
| summarize AvgCPUUsageNanoCores = avg(CounterValue) by bin(TimeGenerated, 30m), InstanceName, _ResourceId
```

So it would be great if we could build a query for the Azure client, then schedule the scrape and scrape those metrics the same way we did for Azure Monitor's Metrics.

Azure Monitor Query Logs client reference: https://docs.microsoft.com/en-us/dotnet/api/azure.monitor.query.logsqueryclient.-ctor?view=azure-dotnet#azure-monitor-query-logsqueryclient-ctor

**Comment by tomkerkhove** [comment]
That's a good idea! Feel free to open a feature request for an `azure-log-analytics-query` scraper or so.

Any thoughts if you want to contribute this?
